src/Functions/reinterpretAs.cpp

```diff
@@ -255,18 +264,47 @@ class FunctionReinterpret : public IFunction
 
                     static constexpr size_t copy_size = std::min(sizeof(From), sizeof(To));
 
-                    for (size_t i = 0; i < size; ++i)
+                    if constexpr (std::is_same_v<FromType, DataTypeUUID>)
                     {
-                        if constexpr (std::endian::native == std::endian::little)
-                            memcpy(static_cast<void*>(&to[i]), static_cast<const void*>(&from[i]), copy_size);
-                        else
+                        size_t i = 0;
+                        std::ranges::for_each(
+                            from,
+                            [&](auto value)
+                            {
+                                // Transform the UUID copy prior to copying to the target.
+                                transformEndianness<std::endian::little>(value);
+                                UUIDHelpers::toLegacyFormat(value);
```

Comment:
I think it's just these two lines that have to be performed in case of little-endian, so you could drop lines 267-290 in favour of putting in a special case in at around what's currently line 295-296 and always make a copy.